src/components/input-field/input-field.scss

```diff
@@ -131,3 +132,12 @@ input.mdc-text-field__input {
         margin: auto;
     }
 }
+
+.mdc-text-field-character-counter {
+  transition: opacity 0.2s ease 0.5s;
```

Comment:
Nice, but I find it pretty annoying that the helper-line doesn't have the same transition. We should either add it there too, or remove it from here.

Reply:
Thanks. Well, this was actually intentional. I wanted things to happen with slight delays, so that user can actually follow what is being added to the UI. Because we also have the border animation when field gets focused. But it's not hard to remove the `0.5s` delay. 

What do you think?

Reply:
I think we should definitely remove the delay. Animations call attention, and when something happens .5 seconds after I did something, it draws my attention, and that shouldn't be abused for drawing attention to something for no good reason. For no good reason, because more attentive users will notice the character counter even if it's displayed or faded in simultaneously with the border and the helper text, and less attentive users will probably notice it when they start typing and the counter starts ticking up. Even the least attentive users ought to notice it when it hits the max-count, and (presumably) turns red.

Reply:
I think the fading should be used to *lessen* the impact of the showing and hiding of the label, not increase it.

Reply:
fair. I'll fix

Reply:
fixup pushed